### ISSUE TITLE
Batch Processor Feature request

### ISSUE THREAD

**Issue by jinhyunglee** [paninski-lab/yass#164]
It would be nice If I can pass a result from the previous batch process to the next batch process.

**Comment by edublancas** [comment]
I think I understand the motivation for this but let's discuss this on tomorrow's meeting

**Comment by edublancas** [comment]
Just implemented this in #168, here's an example how it works:


```python

    bp = BatchProcessor(path_to_data, dtype='int64', n_channels=2,
                        data_format='long', max_memory='160B')

    def col_sums(data, previous_batch):
        current = np.sum(data, axis=0)
        
       # for the first batch, previous_batch is set to None
       # after that it is set to the result of the previous batch
        if previous_batch is None:
            return current
        else:
            return np.array([previous_batch[0] + current[0],
                             previous_batch[1] + current[1]])

    res = bp.multi_channel_apply(col_sums,
                                 mode='memory',
                                 channels='all',
                                 pass_batch_results=True)


```
